packages/adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -0,0 +1,134 @@
+  /// It is active from a width of 0 to 600 dp and only on mobile devices.
+  static const Breakpoint smallMobile =
+      _Breakpoint(begin: 0, end: 600, platform: _mobile);
+
+  /// This is the defined medium breakpoint. Commonly used to indicate a tablet
+  /// but this breakpoint does not rely on platform.
+  ///
+  /// It is active from a width of 600 to 840 dp.
+  static const Breakpoint medium = _Breakpoint(begin: 600, end: 840);
+
+  /// This is the defined mediumAndUp breakpoint. Starts from small and is open
+  /// ended to include everything onwards
+  ///
+  /// It is active from a width of 600 dp to infinity.
+  static const Breakpoint mediumAndUp = _Breakpoint(begin: 600);
+
+  /// This is the defined medium desktop breakpoint.
+  ///
+  /// It is active from a width of 600 to 840 dp and only on desktop devices.
+  static const Breakpoint mediumDesktop =
+      _Breakpoint(begin: 600, end: 840, platform: _desktop);
+
+  /// This is the defined medium mobile breakpoint.
+  ///
+  /// It is active from a width of 600 to 840 dp and only on mobile devices.
+  static const Breakpoint mediumMobile =
+      _Breakpoint(begin: 600, end: 840, platform: _mobile);
+
+  /// This is the defined large breakpoint. Commonly used to indicate a large
+  /// screen desktop but this breakpoint does not rely on platform.
+  ///
+  /// It is active from a width of 840 dp to infinity.
+  static const Breakpoint large = _Breakpoint(begin: 840);
+
+  /// This is the defined large desktop breakpoint.
+  ///
+  /// It is active from a width of 840 dp to infinity and only on desktop
+  /// devices.
+  static const Breakpoint largeDesktop =
+      _Breakpoint(begin: 840, platform: _desktop);
+
+  /// This is the defined large mobile breakpoint.
+  ///
+  /// It is active from a width of 840 dp to infinity and only on mobile
+  /// devices.
+  static const Breakpoint largeMobile =
+      _Breakpoint(begin: 840, platform: _mobile);
+}
+
+class _Breakpoint extends Breakpoint {
+  const _Breakpoint({this.begin, this.end, this.platform});
+  final double? begin;
+  final double? end;
+  final List<TargetPlatform>? platform;
+
+  @override
+  bool isActive(BuildContext context) {
+    bool size = false;
+    final bool isRightPlatform =
+        platform?.contains(Theme.of(context).platform) ?? true;
+    if (begin != null && end != null) {
+      size = MediaQuery.of(context).size.width >= begin! &&
+          MediaQuery.of(context).size.width < end!;
+    } else if (begin != null && end == null) {
+      size = MediaQuery.of(context).size.width >= begin!;
+    } else if (begin == null && end != null) {
+      size = MediaQuery.of(context).size.width < end!;
+    }
+    return size && isRightPlatform;
+  }
+}
+
+/// A class to indicate whether a given number of conditions based on the
+/// current [BuildContext] are met or not. This class can really be used for
+/// wide variety of purposes but the typical use is within [SlotLayout] and
+/// [AdaptiveScaffold].
```

Comment:
```suggestion
/// An interface to define the conditions that distinguish between types of screens.
///
/// Adaptive apps usually display differently depending on the screen type: a compact
/// layout for smaller screens, or a relaxed layout for larger screens. Override this class
/// by defining `isActive` to fetch the screen property (usually `MediaQuery.of`) and
/// return true if the condition is met.
///
/// Typically, it's not needed to keep breakpoints exclusive between each other, since
/// they are tested one by one with a pre-defined priority.
///
/// If the condition is only based on the screen width and/or the device type, use [WidthBreakpoint].
///
/// See also:
///
///  * [SlotLayout.config], which uses breakpoints.
```